### ISSUE TITLE
[@@typescript-eslint/array-type] Object options can cause an error

### ISSUE THREAD

**Issue by maoxiaoke** [typescript-eslint/typescript-eslint#1921]
**Repro**

```JSON
{
  "rules": {
     "@typescript-eslint/array-type": ["warn", { "default": "array-simple" }],
  }
}
```

**Actual Result**

```shell
Error: .eslintrc » xxx:
        Configuration for rule "@typescript-eslint/array-type" is invalid:
        Value {"default":"array"} should be equal to one of the allowed values.
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.28.0` |
| `@typescript-eslint/parser`        | `2.28.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0 |
| `node`                             | `10.8.0` |
| `npm`                              | `6.11.3` |

**Comment by bradzacher** [comment]
I'm unable to repro this. Could you please provide a repro repo?

**Comment by maoxiaoke** [comment]
I make an apology for this. I've used a lower version.

**Comment by Mu-xue** [comment]
@maoxiaoke 请问是什么版本有问题？我也遇到了这个错误
